### ISSUE TITLE
Slowness in filtering dialog because of single-threading

### ISSUE THREAD

**Issue by GiganteX** [xbmc/xbmc#17602]
I am running the Kodi 18.4 (Transparency! UI) connected to a MySQL DB running on a separate machine. The media itself is located on a Synology NAS.
I have about 2.000 movies.

I have a problem with the single-thread behaviour in the filtering UI.

Example:

I click "Filter". First of all, it takes 12 seconds for the filtering dialog to open. Why? You don't need to filter the first thing you do before I have even input anything to filter on!!

So, I plan to filter title on "escape".

I press E... S....

And that's it. Total lockdown of the UI while it filters on "es". Takes a few seconds.

Finally the control is returned to me and the cursor starts blinking again.

I type C... A....

Aaaand total lockdown of the UI while it filters on "esca".... Takes some more seconds.

Well I think you see mny problem.

Realtime filtering as you type is great. The way it's implemented here is not.

Please allow me to type when I want to type, it's very frustrating having to pause while Kodi does work in the background.

**Comment by DaVukovic** [comment]
> Well I think you see mny problem.

Well, I guess yes. The first problem here is, that we don´t see

- which hardware you are using
- which OS you are using
- how big your library is
- which library you are trying to filter
- a full debuglog which would have answered the missing infos above
;)

But what we do see is, that you are using an outdated Kodi

I, personally, can't reproduce your issue on my movie library (about 900 items in there) with Kodi 19 on a CoreI5 6500 running Fedora. Everything is 100% instant during filtering

So without that info (a full debuglog) we can't help at all.

> Please allow me to type when I want to type, it's very frustrating having to pause while Kodi does work in the background.
That ^^ would also be kind of a feature request and I guess many many users are seeing a benefit in real time filtering which seems to be implemented great for the hardware I'm (and probably many others) using. 

So please, get us a log

**Comment by DaveTBlake** [comment]
>Realtime filtering as you type is great. The way it's implemented here is not.

Yes that is true. When pointed at a larger video or music database (one with thousands of items) then the filtering implementation is slow and locks up the UI while it processes. Most of the time is infact lost in storing things in memory, but the filter especialy slow when the db is on MySQL/MariaDB which is must slower than SQLite.

This is a longstanding problem, but no harm to have this issue opened as a reminder. Could be interesting to know some numbers, such as how many items it is filtering, but a debug log won't help much. There is also no quick fix. It requires a significant redesign and there is no timescales for that to happen. Howevere filtering is something I am looking at for the music library, and improvements (when I get there) can pass on to video too in due course.
